File: materias 3 periodo ple/programacao orientada ao objeto/trabalhos praticos/tp-1/Codigo/Cpp/Ordem.cpp
#include "../Hpp/Ordem.hpp"
#include<iostream>


//Implementar get e set Cliente
Ordem :: Ordem(string tipo, string motivo, string descricao, string cpfCliente,double km, bool aprovacao, double valorFinal, bool estado, int codeOrdem):tipo(tipo), motivo(motivo), descricao(descricao), cpfCliente(cpfCliente), km(km), aprovacao(aprovacao), valorFinal(valorFinal), estado(estado), codeOrdem(codeOrdem){}
Ordem :: ~Ordem() {}

//Get e Set para o Tipo da ordem
string Ordem :: getTipo(){
    return tipo;
}

void Ordem :: setTipo(string tipo){
   this->tipo = tipo;
}

//Get e Set para o Motivo da ordem
string Ordem :: getMotivo(){
    return motivo;
}

void Ordem :: setMotivo(string motivo){
    this->motivo = motivo;
}

//Get e Set para a Descricao da ordem
string Ordem :: getDescricao(){
    return descricao;
}

void Ordem :: setDescricao(string descricao){
    this->descricao = descricao;
}

//Get e Set para a km do veiculo
double Ordem :: getKm(){
    return km;
}

void Ordem :: setKm(double km){
    this->km = km;
}

//Get e Set para a aprovacao
bool Ordem :: getAprovacao(){
    return aprovacao;
}

void Ordem :: setAprovacao(bool aprovacao){
    this->aprovacao = aprovacao;
}

//Metodos relacionados ao gerenciamento da lista de serviços
list<Servico> Ordem :: getListaServico(){
    return listaServico;
}

void Ordem :: setListaServico(Servico servico){
    listaServico.push_front(servico);
}

bool Ordem :: removeServico(int codeServico){
    
    for (std::list<Servico>::iterator i = this->listaServico.begin(); i!=this->listaServico.end(); i++){
        if(i->codeServico == codeServico){
            this->listaServico.erase(i);
        }
        else{
            continue;
        }
        
    }     
}

double Ordem :: getValorFinal(){   
    for (std::list<Servico>::iterator i = this->listaServico.begin(); i!=this->listaServico.end(); i++){
        this->valorFinal += i->getValorTotal();
        
    }  
    return valorFinal;
}

//Get e Set para o estado da Ordem(finalizada ou em andamento)
bool Ordem :: getEstado(){
    return estado;
}

void Ordem :: setEstado(bool estado){
    this->estado = estado;
}

int Ordem :: getCodeOrdem(){
    return codeOrdem;
}

void Ordem :: setCodeOrdem(int codeOrdem){
    this->codeOrdem = codeOrdem;
}

string Ordem :: getCpfCliente() {
    return cpfCliente;
}

void Ordem :: setCpfCliente(string cpfCliente) {
    this->cpfCliente = cpfCliente;
}

File: materias 3 periodo ple/programacao orientada ao objeto/trabalhos praticos/tp-2/SistemaOficina/src/sistemaoficina/Servicos/Ordem.java
package sistemaoficina.Servicos;

import java.util.ArrayList;
import java.util.List;

public class Ordem{
	private String tipo;
	private String motivo;
	private String descricao;
	private double km;
	private boolean aprovacao;
	private double valorFinal;
	private boolean estado;
	private List<Servico> listaServico;
	private int codeOrdem;
	private String cpfCliente;
	
        //Construtos basico da Ordem responsavel por inicializar apenas a lista Servico
	public Ordem(){
            
		List<Servico> listaServico = new ArrayList<Servico>();
	}
        //Construtor complexo da Ordem responsavel por inicializar os campos da Ordem com os parametros recebidos
	public Ordem(String tipo, String motivo, String descricao, double km, boolean aprovacao, boolean estado, int codeOrdem, String cpfCliente){
		setTipo(tipo);
		setMotivo(motivo);
		setDescricao(descricao);
		setKm(km);
		setAprovacao(aprovacao);
		listaServico = new ArrayList<Servico>();
		setCodeOrdem(codeOrdem);
		setCpf(cpfCliente);
                valorFinal = 0;
	}
	
	//Sets e Gets
	
	//Get e Set Tipo
        
	public void setTipo(String tipo){
		this.tipo = tipo;
	}
	
	public String getTipo(){
		return tipo;
	}
	//Get e Set Motivo
	public void setMotivo(String motivo){
		this.motivo = motivo;
	}
	
	public String getMotivo(){
		return motivo;
	}
	
	//Get e Set Descricao
	public void setDescricao(String descricao){
		this.descricao = descricao;
	}
	
	public String getDescricao(){
		return descricao;
	}
	
	//Get e Set Km
	public void setKm(double km){
		this.km = km;
	}
	public double getKm(){
		return km;
	}
	
	//Get e Set Aprovacao
	public void setAprovacao(boolean aprovacao){
            this.aprovacao = aprovacao;
	}
	
	public boolean getAprovacao(){
            return aprovacao;
	}
	
	//Get Valor Final
	public double getValorFinal(){
            Servico servico = new Servico();		
            for(int i = 0; i < listaServico.size(); i++){
                servico = listaServico.get(i);
                valorFinal += servico.getValorTotal();
            }
            return valorFinal;
	}
	
	//Get e Set Estado	
	public void setEstado(boolean estado){
            this.estado = estado;
	}
	
	public boolean getEstado(){
            return estado;
	}
	
	//Get e Set Servico
        public void setServico(Servico servico){
            listaServico.forEach(serv->{ if(serv.getCodeServico() == serv.getCodeServico()) serv = servico; });        
        }
        
	public boolean addServico(Servico servico){
            if(!listaServico.contains(servico)) {
            
                listaServico.add(servico);
           
            return true;
        }
        
        return false;
    }
	
    //Funcao que remove um servico da lista   
    public void removeServico(Servico servico){
        //Verifica se a lista esta vazia ou nao
        if(!listaServico.isEmpty())
           listaServico.remove(servico);
    }
    //Funcao remove um servico da lista baseado no codigo recebido
    public void deleteServico(int codeServico){
        //Loop percorrendo o vetor        
        for(int i = 0; i < listaServico.size(); i++){
            //Caso o servico contenha o codigo identico ao desejado ele e removido da lista servico
            if(listaServico.get(i).getCodeServico() == codeServico){
                //Chamando o metodo para retirar uma lista servico
                removeServico(listaServico.get(i));
            }
            else continue;
        }
    }
                
    public Servico getServico(int codeServico){
        for(Servico servico : listaServico) {
            if(servico.getCodeServico() == codeServico)
                return servico;
        }

            return null;           
              
    }
    public List<Servico> getListaServico(){
    
        return listaServico;
    }
    
	
    //Get e Set Code Ordem
    public void setCodeOrdem(int codeOrdem){
        this.codeOrdem = codeOrdem;
    }
	
    public int getCodeOrdem(){
	return codeOrdem;
    }
	
    //Get e Set Cpf
    public void setCpf(String cpf){
	this.cpfCliente = cpf;
    }
	
    public String getCpf(){
        return cpfCliente;
    }
	
        @Override
    public boolean equals(Object o) {
        if(this == o)
            return true;
        
        if(o == null)
            return false;
        
        if(getClass() != o.getClass())
            return false;

        Ordem ordem = (Ordem)o;
        
        return (this.getCodeOrdem() == ordem.getCodeOrdem() && this.getCpf() == ordem.getCpf());
    }
}